src/main/scala/analysis/Local.scala

```diff
@@ -0,0 +1,351 @@
+        val lhsCell = graph.adjust(graph.varToCell(n)(variable))
+        var global = isGlobal(rhs, n)
+        var stack = isStack(rhs, n)
+        if global.isDefined then // Rx = global address
+          graph.mergeCells(lhsCell, global.get)
+        else if stack.isDefined then // Rx = stack address
+          graph.mergeCells(lhsCell, stack.get)
+        else
+          expr match
+            case BinaryExpr(op, arg1: Variable, arg2) => // Rx = Rx + c
+              val arg2Offset = evaluateExpression(arg2, constProp(n))
+              if op.equals(BVADD) && arg1.equals(stackPointer)
+              && arg2Offset.isDefined && isNegative(arg2Offset.get) then
+                () // the stack is handled prior to this
+//                val size = bv2SignedInt(arg2Offset.get)
+//                val node = DSN(Some(graph))
+//                node.allocationRegions.add(StackLocation("Stack_"+proc.name, proc, -size))
+//                node.flags.stack = true
+//                graph.mergeCells(lhsCell, node.cells(0))
+              else if /*varToSym.contains(n) &&  varToSym(n).contains(arg1) && */ arg2Offset.isDefined then
+                // merge lhs with cell(s) corresponding to (arg1 + arg2) where arg1 is cell and arg2 is an offset
+                val offset = evaluateExpression(arg2, constProp(n)).get.value
+                visitPointerArithmeticOperation(n, lhsCell, arg1, 0, false, offset)
+              else // c can't be statically evaluated
+                unsupportedPointerArithmeticOperation(n, expr, lhsCell)
+
+            // Rx = Ry merge corresponding cells to Rx and Ry
+            case arg: Variable /*if varToSym.contains(n) && varToSym(n).contains(arg)*/ =>
+             visitPointerArithmeticOperation(n, lhsCell, arg, 0)
+
+            case MemoryLoad(mem, index, endian, size) => // Rx = Mem[Ry], merge Rx and pointee of Ry (E(Ry))
+              val byteSize = (size.toDouble/8).ceil.toInt
+              lhsCell.node.get.flags.read  = true
+              global = isGlobal(index, n, byteSize)
+              stack = isStack(index, n)
+              if global.isDefined then
+                graph.mergeCells(lhsCell,graph.adjust(graph.find(global.get).getPointee))
+              else if stack.isDefined then
+                graph.mergeCells(lhsCell, graph.adjust(graph.find(stack.get).getPointee))
+              else
+                index match
+                  case BinaryExpr(op, arg1: Variable, arg2) =>
+                    evaluateExpression(arg2, constProp(n)) match
+                      case Some(v) =>
+//                        assert(varToSym(n).contains(arg1))
+                        val offset = v.value
+                        visitPointerArithmeticOperation(n, lhsCell, arg1, byteSize, true, offset)
+                      case None =>
+//                        assert(varToSym(n).contains(arg1))
+                        visitPointerArithmeticOperation(n, lhsCell, arg1, byteSize, true, 0, true)
```

Comment:
Is it correct to just assume that the value of arg2 is 0 if it couldn't be evaluated?

Reply:
https://github.com/UQ-PAC/BASIL/pull/222#discussion_r1770753140

Reply:
I think it makes sense if this case is unsupportedPointerArithmeticOperation since arg2 could not be evaluated?